packages/sdk/test/end-to-end/contract-call-cache.test.ts

```diff
@@ -0,0 +1,100 @@
+import { config as CHAIN_CONFIG } from '@streamr/config'
+import { fetchPrivateKeyWithGas } from '@streamr/test-utils'
+import { collect, StreamID, until } from '@streamr/utils'
+import { Wallet } from 'ethers'
+import { StreamPermission } from '../../src/permission'
+import { StreamrClient } from '../../src/StreamrClient'
+import { ProxyHttpServer, ProxyHttpServerRequest } from '../test-utils/ProxyHttpServer'
+import {
+    createRelativeTestStreamId,
+    createTestClient,
+    createTestStream,
+    formEthereumFunctionSelector,
+    parseEthereumFunctionSelectorFromCallData
+} from '../test-utils/utils'
+import { nextValue } from './../../src/utils/iterators'
+
+export const waitForTheGraphToHaveIndexed = async (streamId: StreamID, client: StreamrClient): Promise<void> => {
+    await until(async () => {
+        const streams = await collect(client.searchStreams(streamId, undefined))
+        return streams.length > 0
+    })
+}
+
+describe('contract call cache', () => {
+
+    describe('metadata', () => {
+
+        let client: StreamrClient
+        let authenticatedUser: Wallet
+        let server: ProxyHttpServer
+        let existingStreamId: StreamID
+        const METADATA_QUERY_FUNCTION_SELECTOR = formEthereumFunctionSelector('getStreamMetadata(string)')
+
+        const getMethodCalls = (): ProxyHttpServerRequest[] => {
+            const methodCalls = server.getRequests().filter(((r) => r.body.method === 'eth_call'))
+            return methodCalls.filter((c) => {
+                return parseEthereumFunctionSelectorFromCallData(c.body.params[0].data) === METADATA_QUERY_FUNCTION_SELECTOR
+            })
+        }
+
+        beforeAll(async () => {
+            authenticatedUser = new Wallet(await fetchPrivateKeyWithGas())
+            const creator = createTestClient(await fetchPrivateKeyWithGas())
+            existingStreamId = (await createTestStream(creator, module)).id
+            creator.grantPermissions(existingStreamId, {
+                userId: authenticatedUser.address,
+                permissions: [StreamPermission.EDIT]
+            })
+            await creator.destroy()
+            await waitForTheGraphToHaveIndexed(existingStreamId, creator)
+        })
+
+        beforeEach(async () => {
+            server = new ProxyHttpServer(CHAIN_CONFIG.dev2.rpcEndpoints[0].url)
```

Comment:
FWIW we could also test this same behavior without an actual server if we were able to somehow mock the `streamRegistryContractReadonly` instance within `StreamRegistry`.

Reply:
Yes, then it would be unit test.